packages/odyssey-react-mui/src/labs/OdysseyPickers/ComposablePicker.tsx

```diff
@@ -0,0 +1,191 @@
+    OptionType,
+    HasMultipleChoices,
+    undefined,
+    IsCustomValueAllowed
+  >["renderOption"];
+
+  renderTags?: MuiAutocompleteProps<
+    OptionType,
+    HasMultipleChoices,
+    undefined,
+    IsCustomValueAllowed
+  >["renderTags"];
+};
+
+const ComposablePicker = <
+  OptionType,
+  HasMultipleChoices extends boolean | undefined,
+  IsCustomValueAllowed extends boolean | undefined,
+>({
+  ariaDescribedBy,
+  defaultValue,
+  errorMessage,
+  errorMessageList,
+  getIsOptionEqualToValue,
+  getOptionLabel,
+  groupOptionsBy,
+  hasMultipleChoices,
+  id: idOverride,
+  inputValue,
+  isCustomValueAllowed,
+  isDisabled,
+  isFullWidth = false,
+  isLoading,
+  isOptional = false,
+  isReadOnly,
+  isVirtualized: isVirtualizedProp = false,
+  hint,
+  HintLinkComponent,
+  label,
+  name: nameOverride,
+  onBlur,
+  onChange: onChangeProp,
+  onInputChange: onInputChangeProp,
+  onFocus,
+  options,
+  renderOption,
+  renderTags,
+  value,
+  testId,
+  translate,
+}: ComposablePickerProps<
+  OptionType,
+  HasMultipleChoices,
+  IsCustomValueAllowed
+>) => {
+  const {
+    inputValueProp,
+    isVirtualized,
+    onChange,
+    onInputChange,
+    renderInput,
+    valueProps,
+    ListboxComponent,
+  } = useAutocomplete<OptionType, HasMultipleChoices, IsCustomValueAllowed>({
+    ariaDescribedBy,
+    defaultValue,
+    errorMessage,
+    errorMessageList,
+    hasMultipleChoices,
+    hint,
+    HintLinkComponent,
+    inputValue,
+    isFullWidth,
+    isOptional,
+    isVirtualized: isVirtualizedProp,
+    label,
+    name: nameOverride,
+    onChange: onChangeProp,
+    onInputChange: onInputChangeProp,
+    testId,
+    value,
+  });
+
+  return (
+    <MuiAutocomplete
+      {...valueProps}
+      {...inputValueProp}
+      // conditionally provide the ListboxComponent if this needs to be virtualized
+      {...(isVirtualized.current && { ListboxComponent })}
+      // AutoComplete is wrapped in a div within MUI which does not get the disabled attr. So this aria-disabled gets set in the div
+      aria-disabled={isDisabled}
+      disableCloseOnSelect={hasMultipleChoices}
+      disabled={isDisabled}
+      freeSolo={isCustomValueAllowed}
+      filterSelectedOptions={true}
+      fullWidth={isFullWidth}
+      getOptionLabel={getOptionLabel}
+      groupBy={groupOptionsBy}
+      id={idOverride}
+      isOptionEqualToValue={getIsOptionEqualToValue}
+      loading={isLoading}
+      multiple={hasMultipleChoices}
+      onBlur={onBlur}
+      onChange={onChange}
+      onInputChange={onInputChange}
+      onFocus={onFocus}
+      options={options}
+      readOnly={isReadOnly}
+      renderInput={renderInput}
+      renderOption={renderOption}
+      renderTags={renderTags}
+      translate={translate}
+      open={true}
+    />
+  );
+};
+
+// Need the `typeof Autocomplete` because generics don't get passed through
+const MemoizedComposablePicker = memo(
+  ComposablePicker,
+) as typeof ComposablePicker;
```

Comment:
```suggestion
// Need `typeof ComposablePicker` because generics don't get passed through
const MemoizedComposablePicker = memo(
  ComposablePicker,
) as typeof ComposablePicker;
```